dotcom-rendering/docs/contributing/how-to.md

```diff
@@ -25,3 +37,41 @@ If you need JavaScript on the client, there are two main options:
 
 Both of these approaches should be used with caution, though. Feel free to reach
 out to the team if you're working on a feature in DCR which needs client-side scripts!
+
+## How can I create an 'island' in DCR?
+
+The 'islands' pattern is DCR's way of adding client-side React code to the site.
+You can read more about it in our
+[Improved Partial Hydration](architecture/027-better-partial-hydration.md)
+decision document.
+
+To add an island:
+
+1. Wrap your component on the server with an `<Island>` component.
+2. Add `.importable` to the component filename. Eg: `[MyThing].importable.tsx`
+3. Specify what should trigger hydration (e.g. waiting until the component
+  scrolls into view). See `Island.tsx` props for options.
+
```

Comment:
If you wrap these steps into code examples - would that give even more detail with less reading? Same for async requests.

Reply:
I'd be very happy to add code examples, but I avoided doing so here because:

i) it seems likely that code samples will go stale quicker than the general pattern we're using, and 
ii) a fuller description allows us to make non-code conventions explicit (i.e. in this case point (2) is crucial but it's also relatively hard to spot in the island code).

What do you think?